python/cudf/cudf/core/dataframe.py

```diff
@@ -5263,40 +5264,67 @@ def quantile(
 
         Returns
         -------
+        Series or DataFrame
+            If q is an array or numeric_only is set to False, a DataFrame
+            will be returned where index is q, the columns are the columns
+            of self, and the values are the quantile.
 
-        DataFrame
+            If q is a float, a Series will be returned where the index is
+            the columns of self and the values are the quantiles.
+
+        Notes
+        -----
+        One notable difference from Pandas is when DataFrame is of
+        non-numeric types and result is expected to be a Series in case of
+        Pandas. cuDF will return a DataFrame as it doesn't support mixed
+        types under Series yet.
```

Comment:
We haven't committed to ever supporting mixed types:
```suggestion
        types under Series.
```